src/components/RecordTxDialog/RecordTxDialog.tsx

```diff
@@ -0,0 +1,109 @@
+import { QuestionMarkCircleIcon as Question } from '@heroicons/react/24/outline';
+import { ArchiveRestore } from 'lucide-react';
+
+import React, { useState } from 'react';
+
+import { toast } from 'sonner';
+
+import { Button } from '@/components/ui/button.tsx';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog.tsx';
+import { Input } from '@/components/ui/input.tsx';
+import { Label } from '@/components/ui/label.tsx';
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from '@/components/ui/tooltip.tsx';
+import { useSpvWalletClient } from '@/contexts';
+import { errorWrapper } from '@/utils';
+
+interface RecordTxDialogProps {
+  className?: string;
+}
+
+export const RecordTxDialog = ({ className }: RecordTxDialogProps) => {
+  const [hexOrId, setHexOrId] = useState<string>('');
+  const { spvWalletClient } = useSpvWalletClient();
+
+  const handleHexOrIdChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setHexOrId(event.target.value);
+  };
+
+  const onSubmit = async () => {
+    if (!hexOrId) return;
+    try {
+      if (hexOrId.length === 64) {
+        const response: any = await fetch(`https://api.whatsonchain.com/v1/bsv/main/tx/${hexOrId}/hex`);
+        const hex = await response.text();
+        await spvWalletClient!.AdminRecordTransaction(hex);
+
+        toast.success('Recorded Transaction');
+        setHexOrId('');
+        return;
+      } else if (hexOrId.length > 64) {
```

Comment:
If you do "return" you don't have to use "else" below. 
Or - if you want to use "if..else", skip the return part

Reply:
You're right. Thanks